instrument-starters/opentracing-spring-cloud-jdbc-starter/src/test/java/io/opentracing/contrib/spring/cloud/jdbc/jdbc/JdbcOnlyWithActiveTest.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.opentracing.contrib.spring.cloud.jdbc.jdbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import io.opentracing.Scope;
+import io.opentracing.contrib.spring.cloud.jdbc.MockTracingConfiguration;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Optional;
+import javax.sql.DataSource;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Test behaviour when withActiveSpanOnly is set
+ * @author Will Penington
+ */
+@SpringBootTest(classes = {MockTracingConfiguration.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+@TestPropertySource(properties = {
+        "opentracing.spring.cloud.jdbc.withActiveSpanOnly=true"
```

Comment:
Could we have the same test with the default configuration?

Reply:
Added.

Reply:
So I copied this test and removed setting the property to let it be the default. The test itself passed however when I ran all the tests together they failed with this error:
```
Failed tests:   spanJoinsActiveSpan(io.opentracing.contrib.spring.cloud.jdbc.JdbcTracingTest): expected:<1> but was:<0>
  spanIsCreatedWhenUsingJdbcTemplate(io.opentracing.contrib.spring.cloud.jdbc.JdbcTracingTest): expected:<1> but was:<0>
  spanIsCreatedForPreparedStatement(io.opentracing.contrib.spring.cloud.jdbc.JdbcTracingTest): expected:<1> but was:<0>

Tests in error: 
  concurrentParents(io.opentracing.contrib.spring.cloud.jdbc.JdbcTracingTest): Condition with lambda expression in io.opentracing.contrib.spring.cloud.jdbc.JdbcTracingTest was not fulfilled within 10 seconds.
```
I went through a bunch of iterations to see what the exact issue is as you can see through all the commits and reverts. The only thing I can pin point is that having a third test that sets properties different than this one, the test in the `JdbcTracingTest` fails with said error. I realized that this exact test is in `JdbcTracingTest` (https://github.com/opentracing-contrib/java-spring-cloud/blob/master/instrument-starters/opentracing-spring-cloud-jdbc-starter/src/test/java/io/opemtracing/contrib/spring/cloud/jdbc/JdbcTracingTest.java#L73:L81) with the default values so I believe that test covers this case. I will leave it that because I cannot get the tests to pass with adding the defaults test.